src/render/decorations/CHyprGroupBarDecoration.cpp

```diff
@@ -364,11 +387,13 @@ bool CHyprGroupBarDecoration::onBeginWindowDragOnDeco(const Vector2D& pos) {
 }
 
 bool CHyprGroupBarDecoration::onEndWindowDragOnDeco(const Vector2D& pos, PHLWINDOW pDraggedWindow) {
+    static auto PSTACKED = CConfigValue<Hyprlang::INT>("group:groupbar:stacked");
     if (!pDraggedWindow->canBeGroupedInto(m_pWindow.lock()))
         return false;
 
     const float BARRELATIVEX = pos.x - assignedBoxGlobal().x - m_fBarWidth / 2;
-    const int   WINDOWINDEX  = BARRELATIVEX < 0 ? -1 : (BARRELATIVEX) / (m_fBarWidth + BAR_HORIZONTAL_PADDING);
```

Comment:
we should make sure this is to -1 if the value is negative so you can place it at the most left order (otherwise it might get rounded to 0 which will place it in the next place)

Reply:
Slightly reworked all this stuff and it seems to behave properly in my testing